content/hardware/04.pro/boards/portenta-x8/tutorials/waves-fleet-managment/content.md

```diff
@@ -0,0 +1,111 @@
+description: 'Learn how to use Foundries.io Factory fleet management tool Waves to manage multiple Portenta X8 devices'
+difficulty: intermediate
+tags:
+  - Embedded Linux
+  - Flashing
+  - Foundries.io
+author: 'Benjamin Dannegård'
+hardware:
+  - hardware/04.pro/boards/portenta-x8
+---
+
+## Overview
+
+In a production environment it is convenient to plan updates, and have control over when and which devices are updated. FoundriesFactory Waves is the feature for this. It allows you to easily define a group of Portenta X8s and then push updates to that specific group. This tutorial will show you how to define that group and how to construct a Wave that can then be pushed to a group.
+
+## Goals
+
+- Learn how to use Waves fleet manager
+- Learn how to assign a target to a Wave
+- Learn how to push a Wave to a group of devices
+
+### Required Hardware and Software
+
+- USB-C to USB-A or USB-C to USB-C
+- Portenta X8
+- Arduino Create account
+- Arduino Pro Cloud Subscription. [Learn more about the Pro Cloud](https://www.arduino.cc/pro/hardware/product/portenta-x8#pro-cloud).
+- Foundries.io account (linked with the Pro Cloud subscription)
+- FoundriesFactory® ([Check the Getting Started tutorial](https://docs.arduino.cc/tutorials/portenta-x8/out-of-the-box))
+- Devices already attached to your factory ([Check the Getting Started tutorial](https://docs.arduino.cc/tutorials/portenta-x8/out-of-the-box))
+    
+## Instructions
+
+### Setting Up the Terminal
+
+Waves fleet management requires us to have the X8 setup with a FoundriesFactory. If you have not done this please take a look our other tutorial [Getting Started tutorial](https://docs.arduino.cc/tutorials/portenta-x8/out-of-the-box), it will walk you through how to set up the X8 with your FoundriesFactory. To use Waves we also need to have fioctl installed and configured, you can follow this guide [here](https://docs.foundries.io/latest/getting-started/install-fioctl/index.html) for setting up fioctl. Creating waves and device groups will be done via the host, which is your factory, so the following commands will be in a terminal using fioctl connected to your FoundriesFactory.
+
+### Rotating Our Keys
+
+For security purposes we recommend that you rotate the keys of your FoundriesFactory. Rotation of the key will convert the root role's online-key, generated during the bootstrap of a Factory, to an offline key.
+
+First we will rotate the root keys. These are the most important keys, they can be used to create new target keys. We will rotate them with this command:
+```
+fioctl keys rotate-root --initial /absolute/path/to/root.keys.tgz
+```
+
+Now we can rotate the target only keys. With this command:
+```
+fioctl keys rotate-targets /absolute/path/to/root.keys.tgz
+```
+
+And finally we copy the target keys to root using:
+```
+fioctl keys copy-targets /absolute/path/to/root.keys.tgz /path/to/target.only.key.tgz
+```
+
+Now we can move on to creating our Wave.
+
+### Creating the Wave
+
```

Comment:
The initial wave must be:
fioctl wave init -k $PATH_TO_KEY populate-targets <version> <tag>
fioctl wave complete populate-targets
Please check: https://docs.foundries.io/latest/reference-manual/ota/production-targets.html